server/src/lib/data.ts

```diff
@@ -0,0 +1,95 @@
+import {db} from "./db"
+import BlogSchema from "../Schema/Blog"
+import {z} from "zod"
+import { PrismaClient, Category } from '@prisma/client';
+
+
+// getting all the available blogs in the databse 
+export async function getBlog(){
+    return db.blog.findMany()
+}
+
+export async function getBlogByAuthorId(authorId: string) {
+    return db.blog.findMany({
+        where : {
+            authorId,
+        },
+    })
+}
+
+// Create
+async function createAuthor(firstName: string, lastName: string, email: string, password: string): Promise<Author> {
+  return await prisma.author.create({
+    data: {
+      firstName,
+      lastName,
+      email,
+      password,
```

Comment:
password must be encrypted before pushing to database. Look for `bcrypt` library and try to implement it